packages/svelte/src/internal/client/dev/equality.js

```diff
@@ -17,10 +17,11 @@ export function init_array_prototype_warnings() {
 		const index = indexOf.call(this, item, from_index);
 
 		if (index === -1) {
-			const test = indexOf.call(get_proxied_value(this), get_proxied_value(item), from_index);
-
-			if (test !== -1) {
-				w.state_proxy_equality_mismatch('array.indexOf(...)');
+			for (let i = from_index ?? 0; i < this.length; i += 1) {
```

Comment:
can this check as the others that use it be only done in DEV?  Seems kind of wasteful to do in production if an item legitimately doesn't exist.

Reply:
it _is_ only done in dev

Reply:
nice!  oops, sorry, missed that this is all inside `init_array_prototype_warnings` 🤦‍♂️  which is only called in DEV